### PR TITLE
use canonical filename of RTD configuration and enable `nitpicky` mode

### PR DESCRIPTION
**Pull request by zacharyburnett**
### Description

update the name of the ReadTheDocs configuration file to `.readthedocs.yaml` and enable `nitpicky` mode to validate intra-doc links

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -142,3 +142,7 @@
                     'https://www.sdss.org']
 linkcheck_timeout = 180
 linkcheck_anchors = False
+
+# Enable nitpicky mode - which ensures that all references in the docs resolve.
+nitpicky = True
+nitpick_ignore = [('py:obj', 'stsynphot')]
```
Comment (pllim) [inline docs/conf.py:148]: I don't think you have to ignore this anymore since you use double backticks now?
```suggestion
nitpick_ignore = []
```

File: stsynphot/spectrum.py [hunk 2]
```diff
@@ -585,11 +585,11 @@ def ebmvx(redlaw_name, ebv):
 
 def load_vega(vegafile=None, **kwargs):
     """Convenience function to load Vega spectrum that is
-    used throughout `stsynphot`.
+    used throughout ``stsynphot``.
 
     Parameters
     ----------
-    vegafile : str or `None`, optional
+    vegafile : str, None
```
Comment (pllim) [inline stsynphot/spectrum.py:592]: Why this change?

File: stsynphot/spectrum.py [hunk 3]
```diff
@@ -598,7 +598,7 @@ def load_vega(vegafile=None, **kwargs):
 
     Returns
     -------
-    sp : `synphot.spectrum.SourceSpectrum` or `None`
+    sp : synphot.spectrum.SourceSpectrum, None
```
Comment (pllim) [inline stsynphot/spectrum.py:601]: Why? Now it does not link back to the SourceSpectrum API.

https://stsynphot--164.org.readthedocs.build/en/164/api/stsynphot.spectrum.load_vega.html

File: stsynphot/spectrum.py (hunk 2 above)
Comment (zacharyburnett) [reply stsynphot/spectrum.py:592]: It was complaining that `optional` is not a type. I assume that having `None` as an option is indicative of a optional parameter but we can also add that to the description as well

File: stsynphot/spectrum.py (hunk 3 above)
Comment (zacharyburnett) [reply stsynphot/spectrum.py]: Good point, I didn't want it to error but it looks like it works fine with the link

**Review by pllim — APPROVED** [review]
I took the liberty to undo some out-of-scope doc style changes. The PR LGTM now. Thanks for the clean-up!

File: stsynphot/spectrum.py (hunk 2 above)
Comment (pllim) [reply stsynphot/spectrum.py]: Hmm I just saw this. Lemme see if it fails. If it does, I'll add it back in. Thanks!

Comment (pllim) [reply stsynphot/spectrum.py]: p.s. I think RTD is green. 🤷

**Comment by zacharyburnett** [comment]
sounds good, thanks for reviewing!
